The job **`x86_64-gnu-llvm-19`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/14975941711/job/42068259273) [(plain)](https://github.com/rust-lang/rust/commit/b1194d914b79e7f661db2b18e663d0940eb292a1/checks/42068259273/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
#19 exporting to docker image format
#19 sending tarball 20.9s done
#19 DONE 27.2s
##[endgroup]
Setting extra environment values for docker:  --env ENABLE_GCC_CODEGEN=1 --env GCC_EXEC_PREFIX=/usr/lib/gcc/
[CI_JOB_NAME=x86_64-gnu-llvm-19]
[CI_JOB_NAME=x86_64-gnu-llvm-19]
debug: `DISABLE_CI_RUSTC_IF_INCOMPATIBLE` configured.
---
sccache: Listening on address 127.0.0.1:4226
##[group]Configure the build
configure: processing command line
configure: 
configure: build.configure-args := ['--build=x86_64-unknown-linux-gnu', '--llvm-root=/usr/lib/llvm-19', '--enable-llvm-link-shared', '--set', 'rust.randomize-layout=true', '--set', 'rust.thin-lto-import-instr-limit=10', '--set', 'build.print-step-timings', '--enable-verbose-tests', '--set', 'build.metrics', '--enable-verbose-configure', '--enable-sccache', '--disable-manage-submodules', '--enable-locked-deps', '--enable-cargo-native-static', '--set', 'rust.codegen-units-std=1', '--set', 'dist.compression-profile=balanced', '--dist-compression-formats=xz', '--set', 'rust.lld=false', '--disable-dist-src', '--release-channel=nightly', '--enable-debug-assertions', '--enable-overflow-checks', '--enable-llvm-assertions', '--set', 'rust.verify-llvm-ir', '--set', 'rust.codegen-backends=llvm,cranelift,gcc', '--set', 'llvm.static-libstdcpp', '--set', 'gcc.download-ci-gcc=true', '--enable-new-symbol-mangling']
configure: build.build          := x86_64-unknown-linux-gnu
configure: target.x86_64-unknown-linux-gnu.llvm-config := /usr/lib/llvm-19/bin/llvm-config
configure: llvm.link-shared     := True
configure: rust.randomize-layout := True
configure: rust.thin-lto-import-instr-limit := 10
---
failures:

---- [codegen] tests/codegen/option-niche-eq.rs stdout ----

error: verification with 'FileCheck' failed
status: exit status: 1
command: "/usr/lib/llvm-19/bin/FileCheck" "--input-file" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/option-niche-eq/option-niche-eq.ll" "/checkout/tests/codegen/option-niche-eq.rs" "--check-prefix=CHECK" "--allow-unused-prefixes" "--dump-input-context" "100"
stdout: none
--- stderr -------------------------------
/checkout/tests/codegen/option-niche-eq.rs:34:17: error: CHECK-NEXT: is not on the line after the previous match
 // CHECK-NEXT: icmp ult i32
                ^
/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/option-niche-eq/option-niche-eq.ll:25:7: note: 'next' match was here
 %2 = icmp ult i32 %0, %1
      ^
/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/option-niche-eq/option-niche-eq.ll:22:7: note: previous match ended here
start:
      ^
/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/option-niche-eq/option-niche-eq.ll:23:1: note: non-matching line after previous match is here
 %.not = icmp eq i32 %0, 0
^

Input file: /checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/option-niche-eq/option-niche-eq.ll
Check file: /checkout/tests/codegen/option-niche-eq.rs

-dump-input=help explains the following input dump.

Input was:
<<<<<<
         1: ; ModuleID = 'option_niche_eq.c6a99bd56dc48bc5-cgu.0' 
         2: source_filename = "option_niche_eq.c6a99bd56dc48bc5-cgu.0" 
         3: target datalayout = "e-m:e-p270:32:32-p271:32:32-p272:64:64-i64:64-i128:128-f80:128-n8:16:32:64-S128" 
         4: target triple = "x86_64-unknown-linux-gnu" 
         5:  
         6: ; Function Attrs: mustprogress nofree norecurse nosync nounwind nonlazybind willreturn memory(none) uwtable 
         7: define noundef zeroext i1 @non_zero_eq(i32 noundef %0, i32 noundef %1) unnamed_addr #0 { 
         8: start: 
         9:  %_0.sroa.0.0.shrunk = icmp eq i32 %1, %0 
        10:  ret i1 %_0.sroa.0.0.shrunk 
        11: } 
        12:  
        13: ; Function Attrs: mustprogress nofree norecurse nosync nounwind nonlazybind willreturn memory(none) uwtable 
        14: define noundef zeroext i1 @non_zero_signed_eq(i64 noundef %0, i64 noundef %1) unnamed_addr #0 { 
        15: start: 
        16:  %_0.sroa.0.0.shrunk = icmp eq i64 %1, %0 
        17:  ret i1 %_0.sroa.0.0.shrunk 
        18: } 
        19:  
        20: ; Function Attrs: mustprogress nofree norecurse nosync nounwind nonlazybind willreturn memory(none) uwtable 
        21: define noundef zeroext i1 @non_zero_ord(i32 noundef %0, i32 noundef %1) unnamed_addr #0 { 
        22: start: 
        23:  %.not = icmp eq i32 %0, 0 
        24:  %.not4 = icmp ne i32 %1, 0 
        25:  %2 = icmp ult i32 %0, %1 
next:34           !~~~~~~~~~~~         error: match on wrong line
        26:  %_3.sroa.0.0 = select i1 %.not, i1 %.not4, i1 %2 
        27:  ret i1 %_3.sroa.0.0 
        28: } 
        29:  
        30: ; Function Attrs: mustprogress nofree norecurse nosync nounwind nonlazybind willreturn memory(none) uwtable 
        31: define noundef zeroext i1 @non_null_eq(ptr noundef readnone %0, ptr noundef readnone %1) unnamed_addr #0 { 
        32: start: 
        33:  %_0.sroa.0.0.shrunk = icmp eq ptr %1, %0 
        34:  ret i1 %_0.sroa.0.0.shrunk 
        35: } 
        36:  
        37: ; Function Attrs: mustprogress nofree norecurse nosync nounwind nonlazybind willreturn memory(none) uwtable 
        38: define noundef zeroext i1 @ordering_eq(i8 noundef range(i8 -1, 3) %0, i8 noundef range(i8 -1, 3) %1) unnamed_addr #0 { 
        39: start: 
        40:  %_0.sroa.0.0.shrunk = icmp eq i8 %1, %0 
        41:  ret i1 %_0.sroa.0.0.shrunk 
        42: } 
        43:  
        44: ; Function Attrs: mustprogress nofree norecurse nosync nounwind nonlazybind willreturn memory(none) uwtable 
        45: define noundef zeroext i1 @niche_eq(i8 noundef range(i8 0, 8) %0, i8 noundef range(i8 0, 8) %1) unnamed_addr #0 { 
        46: start: 
        47:  %_0.sroa.0.0.shrunk = icmp eq i8 %1, %0 
        48:  ret i1 %_0.sroa.0.0.shrunk 
        49: } 
        50:  
        51: attributes #0 = { mustprogress nofree norecurse nosync nounwind nonlazybind willreturn memory(none) uwtable "probe-stack"="inline-asm" "target-cpu"="x86-64" } 
        52:  
        53: !llvm.module.flags = !{!0, !1} 
        54: !llvm.ident = !{!2} 
        55:  
        56: !0 = !{i32 8, !"PIC Level", i32 2} 
        57: !1 = !{i32 2, !"RtLibUseGOT", i32 1} 
        58: !2 = !{!"rustc version 1.89.0-nightly (01a95aae5 2025-05-12)"} 
>>>>>>
------------------------------------------



```

</details>

--- a/tests/codegen/option-niche-eq.rs
+++ b/tests/codegen/option-niche-eq.rs
@@ -1,3 +1,4 @@
+//@ min-llvm-version: 20
 //@ compile-flags: -Copt-level=3 -Zmerge-functions=disabled
 #![crate_type = "lib"]
 
@@ -24,7 +25,7 @@ pub fn non_zero_signed_eq(l: Option<NonZero<i64>>, r: Option<NonZero<i64>>) -> b
     l == r
 }
 
-// Test for #49892
+// FIXME(#49892)
 // This currently relies on a manual implementation of `PartialOrd`/`Ord` for `Option`
 // Once LLVM is better able to optimize this pattern, we can return to using a derive.
 // CHECK-LABEL: @non_zero_ord
@@ -73,13 +74,3 @@ pub fn niche_eq(l: Option<EnumWithNiche>, r: Option<EnumWithNiche>) -> bool {
     // CHECK-NEXT: ret i1
     l == r
 }
-
-// FIXME: This should work too
-// // FIXME-CHECK-LABEL: @bool_eq
-// #[no_mangle]
-// pub fn bool_eq(l: Option<bool>, r: Option<bool>) -> bool {
-//     // FIXME-CHECK: start:
-//     // FIXME-CHECK-NEXT: icmp eq i8
-//     // FIXME-CHECK-NEXT: ret i1
-//     l == r
-// }

--- a/tests/codegen/option-niche-unfixed/option-bool-eq.rs
+++ b/tests/codegen/option-niche-unfixed/option-bool-eq.rs
@@ -1,0 +1,15 @@
+//@ should-fail
+//@ compile-flags: -Copt-level=3 -Zmerge-functions=disabled
+//! FIXME(#49892)
+//! Tests that LLVM does not fully optimize comparisons of `Option<bool>`.
+//! If this starts passing, it can be moved to `tests/codegen/option-niche-eq.rs`
+#![crate_type = "lib"]
+
+// CHECK-LABEL: @bool_eq
+#[no_mangle]
+pub fn bool_eq(l: Option<bool>, r: Option<bool>) -> bool {
+    // CHECK: start:
+    // CHECK-NEXT: icmp eq i8
+    // CHECK-NEXT: ret i1
+    l == r
+}

--- a/tests/codegen/option-niche-unfixed/option-nonzero-eq.rs
+++ b/tests/codegen/option-niche-unfixed/option-nonzero-eq.rs
@@ -1,0 +1,24 @@
+//@ should-fail
+//@ compile-flags: -Copt-level=3 -Zmerge-functions=disabled
+//! FIXME(#49892)
+//! Test that the derived implementation of `PartialEq` for `Option` is not fully
+//! optimized by LLVM. If this starts passing, the test and manual impl should
+//! be removed.
+#![crate_type = "lib"]
+
+use std::num::NonZero;
+
+#[derive(Copy, Clone, PartialEq, Eq)]
+pub enum Option<T> {
+    None,
+    Some(T),
+}
+
+// CHECK-LABEL: @non_zero_eq
+#[no_mangle]
+pub fn non_zero_eq(l: Option<NonZero<u32>>, r: Option<NonZero<u32>>) -> bool {
+    // CHECK: start:
+    // CHECK-NEXT: icmp eq i32
+    // CHECK-NEXT: ret i1
+    l == r
+}

